File: checkers/cwe/tests/suite/LoopLimitTaintChecker/incorrect_for_loop_struct_subscription.cpp
#include <cstdlib>
#include <fstream>

struct A {
    int* i;
};

void f1()
{
    A a;
    std::ifstream ifs("file");
    ifs >> a.i[5];

    for (int i = 0; i < a.i[5]; i++) {  // Loop here
    }
}

File: checkers/cwe/tests/suite/LoopLimitTaintChecker/java/incorrect_for_loop_struct_subscription.java
import java.util.Scanner;
class A {
    int[] i = new int[10];
}
class B {
    int main()
    {
        Scanner in = new Scanner(System.in);
        A a = new A();
        a.i[5] = in.nextInt();
        for (int i = 0; i < a.i[5]; i++) {  // Loop here
        }
    }
}
